RELEASE_NOTES.rst

```diff
@@ -21,6 +21,570 @@
+  If you do, you should see a warning any time that this connection is retrieved or instantiated (e.g. it should show up in
+  task logs).
+
+  To see if you have any connections that will need to be updated, you can run this command:
+
+  .. code-block:: shell
+
+     airflow connections export - 2>&1 >/dev/null | grep 'non-JSON'
+
+  This will catch any warnings about connections that are storing something other than JSON-encoded Python dict in the ``extra`` field. (#21816)
+
+The ``tree`` default view setting has been renamed to ``grid``
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+  If you set the ``dag_default_view`` config option or the ``default_view`` argument to ``DAG()`` to ``tree`` you will need to update your deployment. The old name will continue to work but will issue warnings. (#22167)
+
+Database configuration moved to new section
+"""""""""""""""""""""""""""""""""""""""""""
+
+  The following configurations have been moved from ``[core]`` to the new ``[database]`` section. However when reading the new option, the old option will be checked to see if it exists. If it does a DeprecationWarning will be issued and the old option will be used instead.
+
+  * sql_alchemy_conn
+  * sql_engine_encoding
+  * sql_engine_collation_for_ids
+  * sql_alchemy_pool_enabled
+  * sql_alchemy_pool_size
+  * sql_alchemy_max_overflow
+  * sql_alchemy_pool_recycle
+  * sql_alchemy_pool_pre_ping
+  * sql_alchemy_schema
+  * sql_alchemy_connect_args
+  * load_default_connections
+  * max_db_retries (#22284)
+
+Remove requirement that custom connection UI fields be prefixed
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+  Hooks can define custom connection fields for their connection type by implementing method ``get_connection_form_widgets``.  These custom fields are presented in the web UI as additional connection attributes, but internally they are stored in the connection ``extra`` dict.  For technical reasons, previously custom field when stored in ``extra`` had to be named with a prefix ``extra__<conn type>__<field name>``.  This had the consequence of making it more cumbersome to define connections outside of the UI, since the prefix is tougher to read and work with. With #22607, we make it so that you can now define custom fields such that they can be read from and stored in ``extra`` without the prefix.
+
+  To enable this, update the dict returned by the ``get_connection_form_widgets`` method to remove the prefix from the keys.  Internally, the providers manager will still use a prefix to ensure each custom field is globally unique, but the absence of a prefix in the returned widget dict will signal to the Web UI to read and store custom fields without the prefix.  Note that this is only a change to the Web UI behavior; when updating your hook in this way, you must make sure that when your *hook* reads the ``extra`` field, it will also check for the prefixed value for backward compatibility. (#22607)
```

Comment:
```suggestion
  To enable the use of short names in the extra dict, in the hook defining the custom fields you must update the dict returned by the ``get_connection_form_widgets`` method to remove the prefix from the keys.  Internally, the providers manager will still use a prefix to ensure each custom field is globally unique, but the absence of a prefix in the returned widget dict will signal to the Web UI to read and store custom fields without the prefix. Note that this is only a change to the Web UI behavior; when updating your hook in this way, you must make sure that when your *hook* reads the ``extra`` field, it will also check for the prefixed value for backward compatibility. (#22607)
```